nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -71,11 +71,7 @@ let
           # anything ever again ("couldn't resolve ..., giving up on
           # it"), so we silently lose time synchronisation. This also
           # applies to openntpd.
-          ${config.systemd.package}/bin/systemctl try-restart ntpd.service
-          ${config.systemd.package}/bin/systemctl try-restart openntpd.service
-
-          # exit code of last command is propagated, so we have to clear it
-          true
+          ${config.systemd.package}/bin/systemctl try-reload-or-restart ntpd.service openntpd.service || true
```

Comment:
I like your solution.
We use systemd-timesyncd without issues. A restart is not necessary.